JPS_BASE_LIB/src/test/java/uk/ac/cam/cares/jps/base/config/test/KeyValueMapTest.java

```diff
@@ -1,250 +1,220 @@
 package uk.ac.cam.cares.jps.base.config.test;
 
-import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.net.URISyntaxException;
 
-import junit.framework.TestCase;
+import org.junit.Assert;
+import org.junit.Test;
 import uk.ac.cam.cares.jps.base.config.KeyValueMap;
 import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
 
-public class KeyValueMapTest extends TestCase {
+public class KeyValueMapTest {
 
-	public void testGetInstance() throws IOException {
+
+	@Test
+	public void testGetInstance() {
 		KeyValueMap kvm = null;
 
 		try {
 			kvm = KeyValueMap.getInstance();
 		} finally {
-			assertNotNull(kvm);
-			assertEquals(kvm.getClass(), KeyValueMap.class);
-			assertEquals(kvm.getClass().getDeclaredFields().length, 7);
-			assertEquals(kvm.getClass().getDeclaredMethods().length, 7);
+			Assert.assertNotNull(kvm);
+			Assert.assertEquals(kvm.getClass(), KeyValueMap.class);
+			Assert.assertEquals(kvm.getClass().getDeclaredFields().length, 7);
+			Assert.assertEquals(kvm.getClass().getDeclaredMethods().length, 7);
 		}
+
 	}
 
+	@Test
 	public void testGet() throws NoSuchMethodException, SecurityException {
 		KeyValueMap kvm = KeyValueMap.getInstance();
-		assertNotNull(kvm.getClass().getDeclaredMethod("get", String.class));
+		Assert.assertNotNull(kvm.getClass().getDeclaredMethod("get", String.class));
 		String key = "key";
 		String val = "val";
 		kvm.put(key, val);
-		assertEquals(kvm.get(key), val);
+		Assert.assertEquals(kvm.get(key), val);
 
+		// Accessing a key with value null should result in an exception
 		try {
-			key = null;
-			kvm.get(key);
+			kvm.get(null);
+			Assert.fail();
```

Comment:
Assert.fail() seems to be redundant here because it will not be asserted when the previous like causes exception which is caught after that and the exception message is asserted instead.

Reply:
The idea is that an exception is expected. Without the Assert.fail(), if no exception is thrown the test will pass, but it should not because we expect an exception. The catch is then to further assert the type of exception and optionally the message.

Reply:
OK, noted. It is like a "belt and suspenders" approach. ✅